### PR TITLE
replace OSM tile server

### LINKED ISSUE #32: Please use preferred tile.openstreetmap.org URL
See: https://github.com/openstreetmap/operations/issues/737

Example:
https://github.com/wrobell/geotiler/blob/b0e13ac3063744f3aeacd973e867cf84db90fe67/geotiler/tests/test_provider.py#L40

### PR DESCRIPTION
**Pull request by lingwhatics**
closes #32
Changes tile server for openstreetmap to new preferred server.
See https://github.com/openstreetmap/operations/issues/737

### CODE REVIEW

File: geotiler/source/osm.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
     "name": "OpenStreetMap",
     "attribution": "© OpenStreetMap contributors\nhttp://www.openstreetmap.org/copyright",
-    "url": "http://{subdomain}.tile.openstreetmap.org/{z}/{x}/{y}.{ext}",
-    "subdomains": ["a", "b", "c"],
+    "url": "http://tile.openstreetmap.org/{z}/{x}/{y}.{ext}",
     "limit": 2
```
Comment (wrobell) [inline geotiler/source/osm.json:5]: We need clarification about the tile usage policy for the limit parameter (see https://github.com/wrobell/geotiler/issues/32#issuecomment-1365891196).

[the remark this answers is not in the document]
Comment (wrobell) [reply geotiler/source/osm.json:5]: We need to keep the limit to `2`, see https://github.com/openstreetmap/operations/issues/737#issuecomment-1366498706.

File: geotiler/tests/test_provider.py [hunk 2]
```diff
@@ -37,17 +37,16 @@ def test_provider_init_default():
     data = {
         'name': 'OpenStreetMap',
         'attribution': '© OpenStreetMap contributors\nhttp://www.openstreetmap.org/copyright',
-        'url': 'http://{subdomain}.tile.openstreetmap.org/{z}/{x}/{y}.{ext}',
+        'url': 'http://tile.openstreetmap.org/{z}/{x}/{y}.{ext}',
```
[the remark this answers is not in the document]
Comment (lingwhatics) [reply geotiler/tests/test_provider.py]: Ok, unit tests for subdomains except in api key replaced.

**Comment by wrobell** [comment]
@lingwhatics Merged, thank you.

**Pull request merged**